src/components/CommentsList/CommentsList.tsx

```diff
@@ -0,0 +1,73 @@
+import {FC, Fragment} from "react";
+import {H1, H4, HorizontalDivider, Spinner, Stack, useDeskproAppTheme} from "@deskpro/app-sdk";
+import {useAdfToAnchoredText, useFindIssueComments} from "../../hooks";
+import { faPlus } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import {useStore} from "../../context/StoreProvider/hooks";
+import {JiraComment} from "../../context/StoreProvider/types";
+import "./CommentsList.css";
+import {ExternalLink} from "../ExternalLink/ExternalLink";
+import ReactTimeAgo from "react-time-ago";
+import truncate from "html-truncate";
+
+interface CommentsListProps {
+    issueKey: string;
+}
+
+export const CommentsList: FC<CommentsListProps> = ({ issueKey }: CommentsListProps) => {
+    const [state, dispatch] = useStore();
+    const { theme } = useDeskproAppTheme();
+    const comments = useFindIssueComments(issueKey);
+    const adfToAnchoredText = useAdfToAnchoredText();
+
+    if (!comments) {
+        return (<Spinner size="small" />);
+    }
+
+    const domain = state.context?.settings.domain as string;
```

Comment:
Given you have a strict invariant here it might be best to have an function assertion.

We have these utility functions in horizon:

```ts
/**
 * Asserts some boolean condition
 * @throws will throw on false condition
 * @example
 * ```
 * assert(foo.type === "bar", "Expected bar");
 * ```
 */
export function assert(condition: boolean, message = "Assertion has failed"): asserts condition {
  if (!condition) {
    throw new Error(message);
  }
}

type Nullish = null | undefined;

/**
 * Asserts that input value is not `null | undefined` -
 * @throws will throw if its either of those values
 * @see notEmpty for a safer alternative
 * @example
 * ```
 * assertDefined(foo, "Expected foo");
 * ```
 */
export function assertDefined<T>(
  val: T | Nullish,
  message = "Assertion has failed"
): asserts val is T {
  if (val === null || val === undefined) {
    throw new Error(message);
  }
}
```
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions
```suggestion
    assertDefined(state.context, "Expected context to be initialised by this point")
```